### PR TITLE
Relax Ruby version constraint

### PR DESCRIPTION
**Pull request by n-rodriguez**
Fix https://github.com/codecov/codecov-ruby/issues/121

### CODE REVIEW

File: codecov.gemspec [hunk 1]
```diff
@@ -12,7 +12,7 @@ Gem::Specification.new do |s|
   s.homepage              = 'https://github.com/codecov/codecov-ruby'
   s.license               = 'MIT'
   s.platform              = Gem::Platform::RUBY
-  s.required_ruby_version = '~> 2.4'
+  s.required_ruby_version = '>= 2.4'
```
Comment (AlexWayfer) [inline codecov.gemspec:15]:
```suggestion
  s.required_ruby_version = '>= 2.4', '< 4'
```

**Review by AlexWayfer — CHANGES_REQUESTED** [review]
Your approach is wrong.

See previous changes and discussion in #109.

Suggestion with the correct code has been provided.

**Comment by n-rodriguez** [comment]
So Rails is wrong too? https://github.com/rails/rails/blob/master/activerecord/activerecord.gemspec#L12

**Comment by AlexWayfer** [comment]
> So Rails is wrong too? https://github.com/rails/rails/blob/master/activerecord/activerecord.gemspec#L12

I think, yes. Do you assume it can't be? Ha-ha. Rails has many errors (see issues), some of them potential (not noticed or not important for now). It's OK, but we can do it better. Rails is not a standard for Ruby (especially out of web).

If you can argue with something except "some gem has this too", but with something truly about Ruby and the subject — please, do.

**Comment by n-rodriguez** [comment]
See the 2019 ruby conf and matz presentation about backward compatibilty. It should br sufficient.

**Comment by AlexWayfer** [comment]
> See the 2019 ruby conf and matz presentation about backward compatibilty. It should br sufficient.

OK, I've checked it. I saw this: https://youtu.be/2g9R7PUCEXo?t=1004

> Ruby 3.0 (?)

So, there were plans for Ruby 2.7 but unknown changes for Ruby 3.0. Because it's unknown feature, and even some changes become reverted before releases.

**Pull request closed without merging**